### DISCUSSION TITLE
Typing issue with uvicorn.run

### DISCUSSION THREAD

**Discussion by AlexanderPodorov** [Kludex/uvicorn#2079] [Potential Issue] [answered]
The following import does not pass type checking by `pyright`:
```python
# shows warning: Type of "run" is partially unknown
from uvicorn import run
```
I've tried to fix the signature with:
1. Replace the bare `typing.Callable` with `typing.Callable[..., typing.Any]`
2. Replace the bare `os.PathLike` with `os.PathLike[str]`

These fixes the issue for me.
However I'm not sure is it a proper fix.

**Comment by Kludex** [comment, marked as answer]
PR welcome to fix it.

**Reply by AlexanderPodorov** [reply]
Are these fixes ok to you?

**Reply by Kludex** [reply]
If it passes the pipeline, I'm okay.
